### ISSUE TITLE
linear-gradient causes infinite loop

### ISSUE THREAD

**Issue by mischnic** [csstools/postcss-preset-env#233]
Input code:
```css
.progress-bar-striped {
  background-image: linear-gradient(
    45deg,
    rgba(255, 255, 255, 0.15) 25%,
    transparent 25%,
    transparent 50%,
    rgba(255, 255, 255, 0.15) 50%,
    rgba(255, 255, 255, 0.15) 75%,
    transparent 75%,
    transparent
  );
}
```

.postcssrc.json
```
{
	"plugins": {
		"postcss-preset-env": true
	}
}
```

Running `postcss input.css` (using https://www.npmjs.com/package/postcss-cli) runs into an infinite loop and at some point the process crashes with OOM.

**Comment by romainmenke** [comment]
Thank you for reporting this.

Can you share the version of `postcss-double-position-gradients` from your lock file?

**Comment by mischnic** [comment]
```yaml
postcss-double-position-gradients@^3.0.1:
  version "3.0.2"
```

I've pushed a complete reproduction to https://github.com/mischnic/parcel-issue-7441

**Comment by romainmenke** [comment]
We were able to replicate and fix the issue.
Will be releasing this as soon as possible.

**Comment by Antonio-Laguna** [comment]
Thank you all again, this has been released as 3.0.3!
